src/main/java/de/tum/in/www1/artemis/web/rest/ProgrammingExerciseResource.java

```diff
@@ -700,4 +710,40 @@ private void validateStaticCodeAnalysisSettings(ProgrammingExercise programmingE
         return updateProgrammingExercise(programmingExercise, null);
     }
 
+    /**
+     * PUT /programming-exercises/:exerciseId/tasks : Parse tasks and corresponding test cases from problem statement on an existing ProgrammingExercise.
+     * @param exerciseId of the exercise
+     * @return the {@link ResponseEntity} with status {@code 200} and with body the tasks with test cases,
+     * or with status {@code 400 (Bad Request) if the exerciseId is not valid}.
+     */
+    @PutMapping(EXTRACT_TASKS)
```

Comment:
This endpoint is a little bit misleading. I changed it to the GET endpoint ProgrammingExerciseTaskResource::getTasks since this describes the actual behavior better: The tasks are loaded from the problem statement.